src/autocomplete.ts

```diff
@@ -1,1644 +1,1654 @@
+            }
+            event.preventDefault();
+            this.forceShowAll = true;
+            this.filterPrep(event, false, true);
+        }, 0);
+    }
+
+    /**
+     * blur behaviour for hiding list and removing focus class(es)
+     * forced when using escape key
+     */
+    handleComponentBlur(event: Event, force: boolean = false) {
+        clearTimeout(this.componentBlurTimer);
+        // use a timeout to ensure this blur fires after other focus events
+        // and in case the user focuses back in immediately
+        const delay: number = force ? 0 : 100;
+        this.componentBlurTimer = setTimeout(() => {
+            // do nothing if blurring to an element within the list
+            const activeElem: Element = document.activeElement;
+            if (
+                !force &&
+                activeElem &&
+                !(this.showAll && this.showAll === activeElem) && // exception for show all button
+                !this.isSelectedElem(activeElem) && // exception for selected items
+                // must base this on the wrapper to allow scrolling the list in IE
+                this.wrapper.contains(activeElem)
+            ) {
+                return;
+            }
+
+            // cancel any running async call
+            if (this.xhr && typeof this.xhr.abort === 'function') {
+                this.xhr.abort();
+            }
+
+            // confirmOnBlur behaviour
+            if (!force && this.options.confirmOnBlur && this.menuOpen) {
+                // if blurring from an option (currentSelectedIndex > -1), select it
+                let toUse: number = this.currentSelectedIndex;
+                if (typeof toUse !== 'number' || toUse === -1) {
+                    // otherwise check for exact match between current input value and available items
+                    toUse = this.indexOfValueIn.call(this, this.filteredSource);
+                }
+                this.handleOptionSelect({}, toUse, false);
+            }
+
+            // cancel timers and hide menu
+            this.cancelFilterPrep();
+            this.hide();
+
+            // in single select case, if current value and chosen value differ, clear selected and input value
+            if (!this.multiple && this.indexOfValueIn.call(this, this.selected) === -1) {
+                if (this.selected.length) {
+                    this.removeEntryFromSelected(this.selected[0]);
+                }
+                const inputOrDdl: boolean = this.elementIsInput || this.elementIsSelect;
+                const originalElement = this.element as HTMLInputElement | HTMLSelectElement;
+
+                // Clear selected and input value ?
+                if(this.options.keepUserInput) {
+                    // set original input value
+                    if (this.elementIsInput && this.input.value !== '') {
+                        originalElement.value = this.input.value;
+                        dispatchEvent(this.element, 'change');
```

Comment:
Shouldn't we trigger the change event on the originalElement instead (same as on line 982) ?